cupy/cuda/memory.pyx

```diff
@@ -1973,3 +1973,28 @@ cdef class PythonFunctionAllocator:
             size, self._malloc_func,
             self._free_func, device.get_device_id())
         return MemoryPointer(mem, 0)
+
+
+cdef dict _allocations = dict()
+
+
+cdef public void* cupy_malloc_ext(
+        ssize_t size, int device_id, void* stream) with gil:
+    # TODO(ecastill) dynamically change stream & device?
+    try:
+        with device.Device(device_id):
+            with stream_module.ExternalStream(ptr=<intptr_t> stream):
+                mem = alloc(size)
+    except Exception:
+        return <void*>0  # return nullptr
+    cdef intptr_t ptr = <intptr_t> mem.ptr
```

Comment:
Is this conversion necessary? I would think the Python dict converts `ptr` back to a Python int so that it can be used as a key?

Reply:
I had lots of compilation errors with the `void*` & `int` related conversions ...
I am not sure if this is strictly necessary now, but will double check